packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -0,0 +1,210 @@
+%lex
+%options case-insensitive
+
+%x string
+%x md_string
+%x title
+%x open_directive
+%x type_directive
+%x arg_directive
+%x close_directive
+%x acc_title
+%x acc_descr
+%x acc_descr_multiline
+%s axis_data
+%s data
+%s data_inner
+%%
+\%\%\{                                    { this.pushState('open_directive'); return 'open_directive'; }
+<open_directive>((?:(?!\}\%\%)[^:.])*)    { this.pushState('type_directive'); return 'type_directive'; }
+<type_directive>":"                       { this.popState(); this.pushState('arg_directive'); return ':'; }
+<type_directive,arg_directive>\}\%\%      { this.popState(); this.popState(); return 'close_directive'; }
+<arg_directive>((?:(?!\}\%\%).|\n)*)      return 'arg_directive';
+\%\%(?!\{)[^\n]*                          /* skip comments */
+[^\}]\%\%[^\n]*                           /* skip comments */
+<axis_data>(\r?\n)                        { this.popState(); return 'NEWLINE'; }
+<data>(\r?\n)                             { this.popState(); return 'NEWLINE'; }
+[\n\r]+                                   return 'NEWLINE';
+\%\%[^\n]*                                /* do nothing */
+
+"title"                                   { return 'title'; }
+
+"accTitle"\s*":"\s*                         { this.pushState("acc_title");return 'acc_title'; }
+<acc_title>(?!\n|;|#)*[^\n]*              { this.popState(); return "acc_title_value"; }
+"accDescr"\s*":"\s*                         { this.pushState("acc_descr");return 'acc_descr'; }
+<acc_descr>(?!\n|;|#)*[^\n]*              { this.popState(); return "acc_descr_value"; }
+"accDescr"\s*"{"\s*                         { this.pushState("acc_descr_multiline");}
+<acc_descr_multiline>[\}]                 { this.popState(); }
+<acc_descr_multiline>[^\}]*               { return "acc_descr_multiline_value"; }
+
+"xychart-beta"                            {return 'XYCHART';}
+("vertical"|"horizontal")                 {return 'CHART_ORIENTATION'}
+
+"x-axis"                                  { this.pushState("axis_data"); return "X_AXIS"; }
+"y-axis"                                  { this.pushState("axis_data"); return "Y_AXIS"; }
+<axis_data>[\[]                           { this.popState(); return 'SQUARE_BRACES_START'; }
+<axis_data>[+-]?(?:\d+(?:\.\d+)?|\.\d+)   { return 'NUMBER_WITH_DECIMAL'; }
```

Comment:
You can add `data_inner` to the state here instead of needing to repeat it like you do below, unless there is a specific reason such as needing to give it lower precedence.
```suggestion
<axis_data,data_inner>[+-]?(?:\d+(?:\.\d+)?|\.\d+)   { return 'NUMBER_WITH_DECIMAL'; }
```

Reply:
@ibrahimWassouf good point. There was `data_inner`, but its action was replacing `data` with `data_inner` rather than pushing new state on top of it. It is my fault, my comments were not clear enough.

To clarify - we are scanning NUMBER_WITH_DECIMAL within 2 different states:
`axis_data` and `data_inner`.
We can leave only one regexp for numbers by assigning a new state:
```
<axis_data>"[" { this.pushState('inner_data'); return 'SQUARE_BRACES_START'; }
<data>"[" { this.pushState('inner_data'); return 'SQUARE_BRACES_START'; }
...
<inner_data>regexp for numbers { return 'NUMBER_WITH_DECIMAL'; }
```

Reply:
Updated the code to reuse the regex, but 
`<axis_data>"[" { this.pushState('inner_data'); return 'SQUARE_BRACES_START'; }`
the inner_data will not work as axis_data the inside data of the [] is a string and for plot data it is numeric.

Reply:
You can push another state if needed, this does not necessarily have to be `inner_data`, that one was just an example